### DISCUSSION TITLE
Is it possible to run without a .nuke root folder?

### DISCUSSION THREAD

**Discussion by SierraNL** [nuke-build/nuke#855] [Q&A]
Is it possible to run without the .nuke folder?

I would like to keep my source tree as clean as possible, and looking at the contents the only import thing in the parameters.json is the path the the solution, can't I set that somewhere (commandline or in the _build.csproj) so I can do without the folder?

**Comment by lonix1** [comment]
Did you find a solution? I'm also interested in this.

I hoped we could customise the `.nuke` directory's location (either to put it inside `build/` or even better, to move it to `/tmp/.nuke`). There's too much stuff in the repo root right now - every tool we use adds it's own `.foo/` directory.

**Comment by ron-myers** [comment]
I checked the code, currently the `.nuke` folder is a constant and not configurable in [Constants.cs](https://github.com/nuke-build/nuke/blob/e4eb7737e08325563ea6ef55f597ac0cd363b1aa/source/Nuke.Common/Constants.cs#L18)

and then Nuke uses that constant to build `GetNukeDirectory` which chains down through the application.  I don't see a way to override this path.

```
internal static AbsolutePath GetNukeDirectory(AbsolutePath rootDirectory)
{
    return rootDirectory / NukeDirectoryName;
}
```

the folder name is also used in Build.ps1 and build.sh.

**Reply by lonix1** [reply]
Thanks for looking into it.

If it's defined once, as a constant, then maybe in the future it will be possible to change it during the setup phase.
